addons/website/models/res_users.py

```diff
@@ -90,8 +90,7 @@ def authenticate(cls, db, login, password, user_agent_env):
                 # visitor records to it.
                 if visitor_pre_authenticate_sudo != visitor_current_user_sudo:
                     visitor_pre_authenticate_sudo._merge_visitor(visitor_current_user_sudo)
-                visitor_current_user_sudo._update_visitor_last_visit()
+                    visitor_current_user_sudo._update_visitor_last_visit()
```

Comment:
@paru-odoo I'm not sure to understand why the indentation here. If the condition is not met, the _update_visitor_last_visit() is not called anymore. Is there a reason?

Reply:
Hello @SergeBayet 
Yes, there is a reason for this change. Consider the following scenario: when a user signs up, their token is assigned a user_partner.id. During the signup process, the authenticate function is called again, at which point it finds an existing visitor_current_user_sudo. In this case, calling _update_visitor_last_visit is unnecessary because the visitor is new and their last connection datetime is already set upon creation.

However, for a returning visitor or one whose session has expired, the visitors need to be merged. During this merge, it is important to update the last connection datetime with the past information. Therefore, we moved the _update_visitor_last_visit call inside the conditional block to ensure it is only called during a merge.

Let me know your POV.
Thanks :)

Reply:
> Let me know your POV. Thanks

@paru-odoo That makes sense. Thank you for your explanation.